### PR TITLE
parser: Don't infinite loop when parsing bogus type longhand

### PR DESCRIPTION
**Pull request by awesomekling**

### CODE REVIEW

File: selfhost/parser.jakt [hunk 1]
```diff
@@ -3564,13 +3564,17 @@ struct Parser {
                             if .previous() is ColonColon {
                                 namespaces.push(namespace_name)
                                 .index++
+                            } else {
+                                .error("Expected ‘::’ before", span)
```
Comment (alimpfard) [inline selfhost/parser.jakt:3568]: before...?

Comment (awesomekling) [reply selfhost/parser.jakt:3568]: Before whatever is highlighted by the span 😅

Comment (alimpfard) [reply selfhost/parser.jakt:3568]: :thonk:
```
.error("Expected `::' here", span: Span(file_id: span.file_id, start: span.start, end: span.start))
```
?

Comment (konradekk) [reply selfhost/parser.jakt:3568]: (format/interpolation string could have a clearer intent here… 🤔 yet it is understandable after moment of though—thatʼs least than two moments, so i acceptable for me! 😁)

**Pull request merged**